src/Azure.Functions.Cli/Common/DurableManager.cs

```diff
@@ -0,0 +1,294 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Interfaces;
+using Colors.Net;
+using DurableTask.AzureStorage;
+using DurableTask.Core;
+using DurableTask.Core.History;
+using Microsoft.Azure.WebJobs.Script;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using static Colors.Net.StringStaticMethods;
+
+namespace Azure.Functions.Cli.Common
+{
+    internal class DurableManager : IDurableManager
+    {
+        private readonly AzureStorageOrchestrationService _orchestrationService;
+
+        private readonly TaskHubClient _client;
+
+        private string TaskHubName;
+
+        private string ConnectionString;
+      
+        public DurableManager(ISecretsManager secretsManager)
+        {
+            this.SetConnectionStringAndTaskHubName(secretsManager);
+
+            var settings = new AzureStorageOrchestrationServiceSettings
+            {
+                TaskHubName = TaskHubName,
+                StorageConnectionString = ConnectionString,
+            };
+
+            _orchestrationService = new AzureStorageOrchestrationService(settings);
+            _client = new TaskHubClient(_orchestrationService);
+        }
+
+        // HELP WANTED: Is there a better way to do this?? (parse host.json for durable settings)
+        private void SetConnectionStringAndTaskHubName(ISecretsManager secretsManager)
+        {       
+            // Set connection string key and task hub name to defaults
+            var connectionStringKey = "AzureWebJobsStorage";
+            TaskHubName = "DurableFunctionsHub";
+
+            try
+            {
+                // Attempt to retrieve Durable override settings from host.json
+                dynamic hostSettings = JObject.Parse(File.ReadAllText(ScriptConstants.HostMetadataFileName));
+
+                if (hostSettings.version?.Equals("2.0") == true)
+                {
+                    // If the version is (explicitly) 2.0, prepend path to 'durableTask' with 'extensions'
+                    connectionStringKey = hostSettings.extensions.durableTask.AzureStorageConnectionStringName ?? connectionStringKey;
+                    TaskHubName = hostSettings.extensions.durableTask.HubName ?? TaskHubName;
+                }
+                else
+                {
+                    connectionStringKey = hostSettings.durableTask.AzureStorageConnectionStringName ?? connectionStringKey;
+                    TaskHubName = hostSettings.durableTask.HubName ?? TaskHubName;
+                }
+            }
+            catch { }
```

Comment:
I recommend at least printing the `Message` property of the exception as a warning (yellow) so that the user can know that something went wrong and report it to us.